ngraph/core/reference/include/ngraph/runtime/reference/fake_quantize.hpp

```diff
@@ -0,0 +1,227 @@
+                Shape out_low_shape(_out_low_shape);
+                Shape out_high_shape(_out_high_shape);
+                std::vector<size_t> arg_memory_offsets(arg_shape.size(), 0);
+                for (int i = arg_shape.size() - 2; i >= 0; i--)
+                {
+                    arg_memory_offsets[i] = std::accumulate(
+                        arg_shape.begin() + i + 1, arg_shape.end(), 1, std::multiplies<size_t>());
+                }
+                align_shape_sizes(in_low_shape, arg_shape.size());
+                align_shape_sizes(in_high_shape, arg_shape.size());
+                align_shape_sizes(out_low_shape, arg_shape.size());
+                align_shape_sizes(out_high_shape, arg_shape.size());
+
+                std::vector<size_t> in_low_offsets, in_high_offsets, out_low_offsets,
+                    out_high_offsets;
+                bool in_low_trivial_broadcast = false;
+                bool in_high_trivial_broadcast = false;
+                bool out_low_trivial_broadcast = false;
+                bool out_high_trivial_broadcast = false;
+                bool in_low_aligned = false;
+                bool in_high_aligned = false;
+                bool out_low_aligned = false;
+                bool out_high_aligned = false;
+
+                auto check_trivial_broadcast =
+                    [&arg_shape, &arg_memory_offsets](Shape& shape_to_check,
+                                                      std::vector<size_t>& target_offsets,
+                                                      bool& trivial_broadcast,
+                                                      bool& aligned) {
+                        if (shape_size(shape_to_check) == 1 || shape_size(shape_to_check) == 0)
+                        {
+                            trivial_broadcast = true;
+                        }
+                        else if (shape_to_check == arg_shape)
+                        {
+                            aligned = true;
+                        }
+                        else
+                        {
+                            target_offsets =
+                                calc_broadcast_index_offset(arg_memory_offsets, shape_to_check);
+                        }
+                    };
+                check_trivial_broadcast(
+                    in_low_shape, in_low_offsets, in_low_trivial_broadcast, in_low_aligned);
+                check_trivial_broadcast(
+                    in_high_shape, in_high_offsets, in_high_trivial_broadcast, in_high_aligned);
+                check_trivial_broadcast(
+                    out_low_shape, out_low_offsets, out_low_trivial_broadcast, out_low_aligned);
+                check_trivial_broadcast(
+                    out_high_shape, out_high_offsets, out_high_trivial_broadcast, out_high_aligned);
+
+                std::vector<size_t> current_dim(arg_shape.size(), 0);
+
+                auto get_value = [&current_dim](bool is_trivial_broadcast,
```

Comment:
`index_offset` can't be calculated once in a common case, since all tensors with quantization ranges can have different shapes, which means that offsets for each of them will be different.